src/data-fetchers/bam/bam-worker.ts

```diff
@@ -359,67 +278,27 @@ const init = (
         // we have to fetch the header before we can fetch data
         bamHeaders[bamUrl] = bamFiles[bamUrl].getHeader();
     }
+    const chromInfo = sizesToChromInfo(chromSizes);
 
-    // if no chromsizes are passed in, we'll retrieve them from the BAM file
-    if (chromSizesUrl) {
-        // cache by bamUrl
-        chromSizes[chromSizesUrl] = fetchChromInfo(chromSizesUrl);
-    }
-
-    dataConfs[uid] = { bamUrl, chromSizesUrl, loadMates, maxInsertSize, extractJunction, junctionMinCoverage };
+    dataConfs[uid] = { bamUrl, chromInfo, loadMates, maxInsertSize, extractJunction, junctionMinCoverage };
 };
 
-const tilesetInfo = (uid: string) => {
-    if (uid in tilesetInfos) {
-        // cache hit
-        return tilesetInfos[uid];
-    }
-    const { chromSizesUrl, bamUrl } = dataConfs[uid];
-    const promises = [
-        // why do we await bamHeaders if not used in this function?
-        bamHeaders[bamUrl],
-        chromSizesUrl ? chromSizes[chromSizesUrl] : undefined
-    ] as const;
-
-    return Promise.all(promises).then(res => {
-        const maybeInfo = res[1];
-        const TILE_SIZE = 1024;
-        let chromInfo: ChromInfo;
-
-        if (maybeInfo) {
-            // this means we received a chromInfo file
-            chromInfo = maybeInfo;
-        } else {
-            // no chromInfo provided so we have to take it from the bam file index
-            const chroms: [name: string, length: number][] = [];
-
-            // @ts-expect-error indexToChr is a protected field
-            const indexToChr = bamFiles[bamUrl].indexToChr;
-
-            for (const { refName: chrName, length } of indexToChr) {
-                chroms.push([chrName, length]);
-            }
-            chroms.sort(natcmp);
-            chromInfo = parseChromsizesRows(chroms);
-        }
-
-        tilesetInfos[uid] = {
-            tile_size: TILE_SIZE,
-            bins_per_dimension: TILE_SIZE,
-            max_zoom: Math.ceil(Math.log(chromInfo.totalLength / TILE_SIZE) / Math.log(2)),
-            max_width: chromInfo.totalLength,
-            min_pos: [0],
-            max_pos: [chromInfo.totalLength],
-            chromInfo: chromInfo
-        };
-
-        return tilesetInfos[uid];
-    });
+const tilesetInfo = async (uid: string) => {
```

Comment:
I saw your comment regarding the speed and just tested on my machine as well. An only issue I found was that a BAM track is not displaying any marks until you interact with them (e.g., use a mouse wheel). I am not sure if this is what you were referring to, but removing `async` and `await` for this function seems to address the issue (I believe these are left by mistake). Other than that, I did not find any speed issues with BAM and VCF files on my machine.

```suggestion
const tilesetInfo = (uid: string) => {
```